File: API/FoodCareApi/Models/Alimento.cs
namespace FoodCareApi.Models;

using System.ComponentModel.DataAnnotations.Schema;

public class Alimento{
        public int idAlimento{ get; set; }
        public int idCategoria{ get; set; }
        public Categoria categoria { get; set;} = null!;
        public int idDoador{ get; set; }
        public Doador doador {get; set;} = null!;
        public string nome  { get; set; } = string.Empty;
        public string descricao { get; set; } = string.Empty;
        public int qntd { get; set; } = 0;
        public DateOnly? validade {get; set;} = default;
        
        [NotMapped]
        public double distancia { get; set; }
     
    }

File: FoodCare/app/src/main/java/com/example/foodcare/model/Alimento.kt
package com.example.foodcare.model

data class AlimentoRequest(
    val nome: String,
    val descricao: String,
    val qntd: Int,
    val validade: String,
    val idCategoria: Int
)

data class AlimentoAtualizarRequest(
    val qntd: Int,
    val validade: String,
    val descricao: String
)

//model para o feed
data class Produto(
    val idAlimento: Int = 0,
    val nome: String = "",
    val idCategoria: Int = 0,
    val idDoador: Int = 0,
    val descricao: String = "",
    val distancia: Double = 0.0,
    val validade: String = "",
    val latitude: Double = 0.0,
    val longitude: Double = 0.0
)

// model para alimentos doador
data class Alimento(
    val idAlimento: Int,
    val nome: String,
    val idCategoria: Int,
    val categoria: String,
    val descricao: String,
    val qntd: Int,
    val validade: String

)